process/resultsParser/resultsParser.go

```diff
@@ -0,0 +1,385 @@
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse outcome on easily found result with return data: %w", err)
+	}
+	if outcome != nil {
+		log.Trace("txHash [%s] result outcome on easily found result with return data", tx.Hash)
+		return outcome, nil
+	}
+
+	outcome, err = parseOutcomeOnTooMuchGasWarning(tx.Logs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse outcome on too much gas warning: %w", err)
+	}
+	if outcome != nil {
+		log.Trace("txHash [%s] result outcome on too much gas warning", tx.Hash)
+		return outcome, nil
+	}
+
+	outcome, err = parseOutcomeOnWriteLogWhereFirstTopicEqualsAddress(tx.Logs, tx.Sender)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse outcome on write log where first topic equals address: %w", err)
+	}
+	if outcome != nil {
+		log.Trace("txHash [%s] result outcome on write log where first topic equals address", tx.Hash)
+		return outcome, nil
+	}
+
+	outcome, err = parseOutcomeWithFallbackHeuristics(tx)
+	if outcome != nil {
+		log.Trace("txHash [%s] result outcome on fallback heuristics", tx.Hash)
+		return outcome, nil
+	}
+	return nil, nil
+}
+
+func parseOutcomeOnSimpleMoveBalance(tx *transaction.ApiTransactionResult) *ResultOutcome {
+	noResults := len(tx.SmartContractResults) == 0
+	var noLogs bool
+	if tx.Logs != nil {
+		noLogs = len(tx.Logs.Events) == 0
+	} else {
+		noLogs = true
+	}
+
+	if noResults && noLogs {
+		return &ResultOutcome{
+			ReturnCode:    "",
+			ReturnMessage: "",
+		}
+	}
+
+	return nil
+}
+
+func parseOutcomeOnInvalidTransaction(tx *transaction.ApiTransactionResult) *ResultOutcome {
+	if tx.Status == transaction.TxStatusInvalid {
+		if tx.Receipt != nil && tx.Receipt.Data != "" {
+
+			return &ResultOutcome{
+				ReturnCode:    "out of funds",
+				ReturnMessage: tx.Receipt.Data,
+			}
+		}
+	}
+
+	// If there's no receipt message, let other heuristics to handle the outcome (most probably, a log with "signalError" is emitted).
+
+	return nil
+}
+
+func parseOutcomeOnEasilyFoundResultWithReturnData(scResults []*transaction.ApiSmartContractResult) (*ResultOutcome, error) {
+	var scr *transaction.ApiSmartContractResult
+	for _, scResult := range scResults {
+		//TODO: check whether scResult.Nonce should be the tx.Nonce + 1
+		if scResult.Nonce != 0 && strings.HasPrefix(scResult.Data, atSeparator) {
+			scr = scResult
+			break
+		}
+	}
+
+	if scr == nil {
+		return nil, nil
+	}
+
+	returnCode, returnDataParts, err := sliceDataFieldInParts(scr.Data)
+	if err != nil {
+		return nil, fmt.Errorf("failed to slice data field in parts: %w", err)
+	}
+
+	returnMessage := string(returnCode)
+	if scr.ReturnMessage != "" {
+		returnMessage = scr.ReturnMessage
+	}
+
+	return &ResultOutcome{
+		ReturnCode:    string(returnCode),
+		ReturnMessage: returnMessage,
+		Values:        returnDataParts,
+	}, nil
+}
+
+func parseOutcomeOnSignalError(logs *transaction.ApiLogs) (*ResultOutcome, error) {
```

Comment:
Function is not used anymore, can be removed.

Reply:
removed